tensorflow/python/keras/layers/normalization.py

```diff
@@ -499,6 +524,37 @@ def _fused_batch_norm(self, inputs, training):
     beta = self.beta if self.center else self._beta_const
     gamma = self.gamma if self.scale else self._gamma_const
 
+    original_shape = None
+    fused_axis = self.axis[0]
+    input_shape = array_ops.shape(inputs)
+
+    # The use of Bessel's correction in training mode imposes the requirement
+    # that the number of elements in the reduced dimensions is > 1.
+    check = control_flow_ops.Assert(
+          training == False or math_ops.reduce_prod(input_shape) > self.depth,
```

Comment:
I think training can also be a tensor. The assert should be added to [this function](https://github.com/tensorflow/tensorflow/blob/807f17221d628b9fc722df5d99562f5ef126842c/tensorflow/python/keras/layers/normalization.py#L552) I think

Reply:
Also probably better to also raise an error immediately if the shape and training boolean are statically known